kb/all-the-ways-to-run-a-cluster.md

```diff
@@ -0,0 +1,123 @@
+---
+slug: all-the-ways-to-run-a-cluster
+title: All the ways to run a Temporal Cluster
+tags:
+  - kb-article
+date: 2022-06-07T00:00:00Z
+---
+
+There are many ways to run a [Temporal Cluster](/clusters) on your own.
+However, the right way for you depends entirely on your use case and where you plan to run it.
+This article aims to maintain a comprehensive list of all the ways we know of.
+
+<!-- truncate -->
+
+## Temporalite
+
+[Temporalite](https://github.com/temporalio/temporalite/cmd/temporalite) is a complete, but minimal, Temporal Cluster distribution (four Temporal Server services and a database) that runs as a single process with zero runtime dependencies.
+It supports persistence to disk and in-memory mode through SQLite.
+
+Temporalite is great for local testing and development purposes.
+It offers great performace in terms of processing Workflow Executions per second.
+Although it does currently default to one History Shard, we plan to make this setting configurable at start time, and you can adjust the setting by cloning the repository.
+In theory, performance at this point is limited to your machine's processing capabilities, SQLite capacity, and SQLite read/write speeds.
+
+One drawback is that Temporalite does not yet support [Advanced Visibility](/visibility/#advanced-visibility), or other Cluster features such as Archival.
```

Comment:
Worth mentioning that we are actively working on bring advanced visibility to SQLite.